grapple/types/redirects.py

```diff
@@ -1,38 +1,76 @@
+import copy
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
+
+from grapple.types.sites import SiteObjectType
 
 from .pages import get_page_interface
 
 
-class RedirectType(graphene.ObjectType):
+class RedirectObjectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
+    site = graphene.Field(SiteObjectType, required=False)
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
```

Comment:
The return type should be `-> str` because `site` is now required, and as such this function will not return `None`. Addressed in cba26966a00bbfb670c390908a84cd91b8cfb4c2